scripts/fix-markdown-violations.sh

```
@@ -0,0 +1,295 @@
+# Function to fix blank lines around fenced code blocks (MD031)
+fix_code_block_spacing() {
+    echo -e "${YELLOW}Fixing code block spacing (MD031)...${NC}"
+    
+    find "$REPO_ROOT" -name "*.md" -type f \
+        ! -path "*/node_modules/*" \
+        ! -path "*/.git/*" \
+        ! -path "*/system-configs/.claude/agents/*.md" \
+        -print0 | while IFS= read -r -d '' file; do
+        
+        # Add blank lines around fenced code blocks
+        if grep -q '^```' "$file"; then
```

Shellcheck (0.10.0)

[error] Couldn't find 'fi' for this 'if'.

(SC1046)